Author: Vikraman Choudhury <vikraman@gentoo.org>
Date:   Fri Jan 24 04:17:16 2014 +0530

    Use --{color,colour} options
    
    --no-{color,colour} is deprecated and removed in eselect 1.4.

    Fix command for listing eselect modules

--- a/_eselect
+++ b/_eselect
@@ -15,7 +15,7 @@ _eselect_env () {
 _eselect_binutils () {
 	local binutilslist
         if (( $words[(I)(set)] )); then
-	  binutilslist=(${${${(M)${(f)"$(eselect --no-color binutils list)"}## *}// \**/}//(  \[*\]   |  \[*\]  )/})
+	  binutilslist=(${${${(M)${(f)"$(eselect --color=no binutils list)"}## *}// \**/}//(  \[*\]   |  \[*\]  )/})
 	  _values "available binutils version" $binutilslist[@] && return 0
         fi
 	_values "binutils options" \
@@ -30,7 +30,7 @@ _eselect_binutils () {
 _eselect_kernel () {
 	local kernellist
         if (( $words[(I)(set)] )); then
-	  kernellist=(${${${(M)${(f)"$(eselect --no-color kernel list)"}## *}// \**/}//(  \[*\]   |  \[*\]  )/})
+	  kernellist=(${${${(M)${(f)"$(eselect --color=no kernel list)"}## *}// \**/}//(  \[*\]   |  \[*\]  )/})
 	  _values "available kernel version" $kernellist[@] && return 0
         fi
 	_values "kenrel options" \
@@ -44,7 +44,7 @@ _eselect_kernel () {
 _eselect_ctags () {
 	local ctagslist
         if (( $words[(I)(set)] )); then
-	  ctagslist=(${${${(M)${(f)"$(eselect --no-color ctags list)"}## *}// \**/}//(  \[*\]   |  \[*\]  )/})
+	  ctagslist=(${${${(M)${(f)"$(eselect --color=no ctags list)"}## *}// \**/}//(  \[*\]   |  \[*\]  )/})
 	  _values "available ctags version" $ctagslist[@] && return 0
         fi
 	_values "ctags options" \
@@ -59,7 +59,7 @@ _eselect_ctags () {
 _eselect_profile () {
 	local profilelist
         if (( $words[(I)(set)] )); then
-	  profilelist=(${${${(M)${(f)"$(eselect --no-color profile list)"}## *}// \**/}//(  \[*\]   |  \[*\]  )/})
+	  profilelist=(${${${(M)${(f)"$(eselect --color=no profile list)"}## *}// \**/}//(  \[*\]   |  \[*\]  )/})
 	  _values -w "available profiles" $profilelist[@] \
 	  "--force[Forcibly set the symlink]" && return 0
         fi
@@ -74,10 +74,10 @@ _eselect_profile () {
 _eselect_fontconfig () {
 	local fclistenabled fclistdisabled
         if (( $words[(I)(enable)] )); then
-	  fclistdisabled=(${${${${(M)${(f)"$(eselect --no-color fontconfig list)"}## *}#*\*}// \**/}//(  \[*\]   |  \[*\]  )/})
+	  fclistdisabled=(${${${${(M)${(f)"$(eselect --color=no fontconfig list)"}## *}#*\*}// \**/}//(  \[*\]   |  \[*\]  )/})
 	  _values -w ".conf to enable" $fclistdisabled[@] && return 0
 	elif (( $words[(I)(disable)] )); then
-	  fclistenabled=(${${${(M)${(M)${(f)"$(eselect --no-color fontconfig list)"}## *}#*\*}// \**/}//(  \[*\]   |  \[*\]  )/})
+	  fclistenabled=(${${${(M)${(M)${(f)"$(eselect --color=no fontconfig list)"}## *}#*\*}// \**/}//(  \[*\]   |  \[*\]  )/})
 	  _values -w ".conf to disable" $fclistenabled[@] && return 0
         fi
 	_values "fontconfig options" \
@@ -91,7 +91,7 @@ _eselect_fontconfig () {
 _eselect_opengl () {
 	local opengllist
 	if (( $words[(I)(set)] )); then
-		opengllist=(${${${(M)${(f)"$(eselect --no-color opengl list)"}## *}// \**/}//(  \[*\]   |  \[*\]  )/})
+		opengllist=(${${${(M)${(f)"$(eselect --color=no opengl list)"}## *}// \**/}//(  \[*\]   |  \[*\]  )/})
 		_values -w "opengl implementations and options" $opengllist[@] \
 		"--use-old[If an implementationis already set, use that one instead]" \
 		"--prefix[Set the source prefix]:path:_files -/" \
@@ -109,7 +109,7 @@ _eselect_opengl () {
 _eselect_vi () {
 	local vilist
 	if (( $words[(I)(set)] )); then
-	  vilist=(${${${(M)${(f)"$(eselect --no-color vi list)"}## *}// \**/}//(  \[*\]   |  \[*\]  )/})
+	  vilist=(${${${(M)${(f)"$(eselect --color=no vi list)"}## *}// \**/}//(  \[*\]   |  \[*\]  )/})
 	  _values -w "vi implementation" $vilist[@] && return 0
 	elif (( $words[(I)(update)] )); then
 	  _values -w "option" "--if-unset[Do not override existing implementation]" && return 0
@@ -127,35 +127,35 @@ _eselect () {
   local globopts sedcmd modnames modopts
 
   globopts=(
-  	"--no-colour[Disable coloured output]" \
-  	"--no-color[Disable coloured output]"
+  	"--colour=no[Disable coloured output]" \
+  	"--color=no[Disable coloured output]"
   		)
-  modnames=(${${${(M)${(f)"$(eselect --no-color list-modules)"}## *}//   */}//  /})
+  modnames=(${${${(M)${(f)"$(eselect --color=no modules list)"}## *}//   */}//  /})
 
   if ((CURRENT == 2)); then
     _arguments -s \
       "$globopts[@]" \
       "*:portage:_values 'eselect modules' \$modnames[@]" && return 0
   elif ((CURRENT == 3)); then
-    if [[ $words[2] == --no-colour || $words[2] == --no-color ]]; then
+    if [[ $words[2] == --colour=no || $words[2] == --color=no ]]; then
       _arguments -s \
         "*:portage:_values 'eselect modules' \$modnames[@]" && return 0
     elif (( $modnames[(I)$words[2]] )); then
       if [[ "$words[2]" == (env|binutils|kernel|ctags|profile|fontconfig|opengl|vi) ]]; then
           _eselect_$words[2] "$@"
       else
-        modopts=(${${${(M)${(f)"$(eselect --no-color $words[2] usage)"}## *}//   */}//  /})
+        modopts=(${${${(M)${(f)"$(eselect --color=no $words[2] usage)"}## *}//   */}//  /})
         _arguments -s \
           "*:portage:_values 'eselect $words[2] options' \$modopts[@]" && return 0
       fi
     fi
   elif ((CURRENT >= 4)); then
-    if (( $words[(I)(--no-color|--no-colour)] )); then
+    if (( $words[(I)(--color=no|--colour=no)] )); then
       if (( $modnames[(I)$words[3]] )); then
         if [[ "$words[3]" == (env|binutils|kernel|ctags|profile|fontconfig|opengl|vi) ]]; then
           _eselect_$words[3] "$@"
         else
-          modopts=(${${${${(M)${(f)"$(eselect --no-color $words[3] usage)"}## *}//   */}//  /}// *})
+          modopts=(${${${${(M)${(f)"$(eselect --color=no $words[3] usage)"}## *}//   */}//  /}// *})
            _arguments -s \
               "*:portage:_values 'eselect $words[3] options' \$modopts[@]" && return 0
         fi
